app/services/katello/applicability/applicable_content_helper.rb

```diff
@@ -45,6 +45,22 @@ def fetch_errata_content_ids
         return Katello::Erratum.find_by_sql([query, {content_facet_id: content_facet.id, repo_ids: self.bound_library_instance_repos} ]).map(&:id).uniq
       end
 
+      def fetch_module_stream_content_ids
+        # Query for all applicable module stream ids
+        query = 'select katello_repository_module_streams.module_stream_id as id from katello_repository_module_streams
+                     inner join katello_module_stream_rpms
+                        on  katello_repository_module_streams.module_stream_id  = katello_module_stream_rpms.module_stream_id
+                     inner join katello_rpms
+                        on katello_rpms.id = katello_module_stream_rpms.rpm_id
+                     inner join katello_content_facet_applicable_rpms on
+                         katello_content_facet_applicable_rpms.rpm_id = katello_rpms.id
+                     where
+                          "katello_content_facet_applicable_rpms"."content_facet_id" = :content_facet_id
+                     AND katello_repository_module_streams.repository_id IN (:repo_ids)'
```

Comment:
nit: I prefer SQL keywords to be uppercase (SELECT, INNER JOIN etc.) but if you don't like that it should at least be consistent within the same SQL statement.

Reply:
If there's ever a migration in the future that changes a table name etc., this would have to be manually updated.  Is this query too difficult to do with Active Record `.joins`, `.ids`, etc?

Reply:
@jeremylenz I can look at changing the query to use more Active Record methods, I don't think we need the manual query necessarily.  I'll change the errata method above if so too.